compiler/src/iree/compiler/Preprocessing/TransformExtensions/PreprocessingExtensions.cpp

```diff
@@ -248,9 +246,8 @@ IREE::transform_dialect::MatchCastCompatibleDagFromRootOp::matchOperation(
 
     // Verify that if already processed, both operations are at the same
     // position.
-    if (targetDagOrder.contains(targetOp) ||
-        payloadDagOrder.contains(payloadOp)) {
-      if (!compareDagOrder(targetOp, payloadOp)) {
+    if (targetToPayloadMapping.contains(targetOp)) {
+      if (targetToPayloadMapping.lookup(targetOp) != payloadOp) {
         return emitSilenceableError() << "dag mismatch";
       }
       continue;
```

Comment:
It won't let me mark line 256 because it's not close enough to these changes, but 3 lines down from this comment there is this comment which needs an update:
```
// Verify general operation equality (name, attributes, regions).
```
Looking at the step 1 comment/changes to the helper.
```
  // Step 1. First just walk from root op "upwards" to match basic
  // producer-consumer match (without checking regions).
```